### ISSUE TITLE
Entity moves over to the next entry at 'fetch_time' instead of 'day_switch_time'

### ISSUE THREAD

**Issue by mal2856** [mampfes/hacs_waste_collection_schedule#950]
I'm not sure if this was intentional, but my entity is dismissing the current day's collections at the fetch_time instead of the day_switch_time.

I haven't set a value for either, so both should be occuring at their default time (fetch at 01:00 and switch at 10:00).

Should I be making sure my fetch_time is after my day_switch_time to stop this from happening?

**Comment by mampfes** [comment]
Could it be that your service provider removes the collections for the current day? This would explain why they disappear at fetch_fetch and not at day_switch_time. You can work around this issue if you set fetch_time and day_switch_time to the same value.

**Comment by Steffie1995** [comment]
I'm having the same problem... My collection days are still in the calendar on the day, just the sensors do not get that there is a collection today...

**Comment by mal2856** [comment]
Thanks. Took a while to be able to test this, but this appears to be the issue.

I solved it by setting the fetch time to the morning, when I want the current day dismissed. I set the day_switch_time to an arbitrary time late in the night after the fetch time.

It might be worth considering changing it so that it automatically fetches more dates at the switch time if there's no new dates available. But otherwise, everything works great.

Thanks for your work on the project, the number of integrations with various local authorities is really impressive.
